### PR TITLE
Implementation of the details page

### PR DESCRIPTION
**Pull request by ezequielsetzesglobant**
The part of the selected contact details screen has been developed.

Video
https://user-images.githubusercontent.com/84338513/163488884-f3797b33-1c13-4235-8fdd-1cab89a9cc01.mov

Test:
Screen Shot 2022-04-14 at 19 31 15

### CODE REVIEW

File: lib/model/contact.dart [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import 'id.dart';
+import 'location.dart';
+import 'name.dart';
+import 'picture.dart';
+import 'registered.dart';
+import 'date_of_birth.dart';
+import 'login.dart';
+
+class Contact {
```
Comment (FranciscoGaraventa) [inline lib/model/contact.dart:9]: We ussualy have Class constructor at the top, then te factory method and last the definition of variables

File: lib/service/contact_service.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import 'dart:convert';
+import 'dart:io';
+import '../model/contact.dart';
+import '../model/contact_data.dart';
+import '../utils/constants.dart';
+import 'interface/i_contact_service.dart';
+import 'package:http/http.dart' as http;
+
+class ContactService extends IContactService {
+  http.Client client = http.Client();
```
Comment (FranciscoGaraventa) [inline lib/service/contact_service.dart]: This is not necessary, instead to define the type of the class, you can replace it with var

File: lib/ui/home_page.dart [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                itemCount: snapshot.data?.length,
+                itemBuilder: (context, index) {
+                  return ListTile(
+                    title: Card(
+                      color: Colors.grey,
+                      child: SizedBox(
+                        height: Constants.contactBoxHeight,
+                        child: Center(
+                          child: Text(
+                            '${snapshot.data?[index].name.first} ${snapshot.data?[index].name.last}',
```
Comment (FranciscoGaraventa) [inline lib/ui/home_page.dart]: This method could be inside Contact class, you can make a method like fullName and here you only need to call that method like snapshot.data?[index].fullName

File: lib/ui/home_page.dart [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                        ),
+                      ),
+                    ),
+                    onTap: () {
+                      Navigator.push(
+                        context,
+                        MaterialPageRoute(
+                          builder: (context) => DetailsPage(
+                            image: '${snapshot.data?[index].picture.large}',
+                            fullName: '${snapshot.data?[index].name.first} ${snapshot.data?[index].name.last}',
```
Comment (FranciscoGaraventa) [inline lib/ui/home_page.dart]: Here you can use the method that a mention before

File: test/details_page_test.dart [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import 'package:contacts_data/ui/details_page.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:network_image_mock/network_image_mock.dart';
+
+void main() {
+  testWidgets('Check for existence of widgets', (WidgetTester tester) async {
+    await mockNetworkImagesFor(
+      () => tester.pumpWidget(const MaterialApp(
+        home: DetailsPage(
```
Comment (FranciscoGaraventa) [inline test/details_page_test.dart]: Remember that you have a json test file for this kind of data

File: lib/ui/details_page.dart [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+import '../utils/text_styles.dart';
+
+class DetailsPage extends StatelessWidget {
+  final String image;
+  final String fullName;
+  final String email;
+  final String phone;
+  final String cell;
+  final String age;
+  final String nationality;
```
Comment (MarcosCGlobant) [inline lib/ui/details_page.dart]: you could send the contact object, to reduce the number of parameters and control the data inside the detailPage itself

File: lib/model/contact.dart (hunk 1 above)
Comment (ezequielsetzesglobant) [reply lib/model/contact.dart:9]: done

File: lib/service/contact_service.dart (hunk 2 above)
Comment (ezequielsetzesglobant) [reply lib/service/contact_service.dart]: done

File: lib/ui/home_page.dart (hunk 3 above)
Comment (ezequielsetzesglobant) [reply lib/ui/home_page.dart]: done

File: lib/ui/home_page.dart (hunk 4 above)
Comment (ezequielsetzesglobant) [reply lib/ui/home_page.dart]: done

File: test/details_page_test.dart (hunk 5 above)
Comment (ezequielsetzesglobant) [reply test/details_page_test.dart]: done

File: lib/ui/details_page.dart (hunk 6 above)
Comment (ezequielsetzesglobant) [reply lib/ui/details_page.dart]: done

**Review by FranciscoGaraventa — APPROVED** [review]

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Review by LeonardoMolleker — APPROVED** [review]

**Pull request merged**